preprocessing-standards.Rmd

```diff
@@ -233,36 +238,49 @@ We perform a t-test to see if there is a significant positive difference between
 samples with high concentrations of standards and samples with low concentration
 
 ```{r}
-# H0: high<=low vs H1: H0^c
-high <- grep("^High",colnames(fmat_pos_watl2)) #c(1,3,6)
-low <- grep("^Low",colnames(fmat_pos_watl2)) #c(2,4,5)
-
-pvalues <- apply(fmat_pos_watl2, 1, function(x) t.test(x[high]-x[low], 
-                        mu = 0, alternative = "greater")$p.value)
-
+# H0: high==low vs H1: H0^c
+high <- grep("^High", colnames(fmat_pos_watl2)) #c(1,3,6)
+low <- grep("^Low", colnames(fmat_pos_watl2)) #c(2,4,5)
+
+pvalues <- apply(fmat_pos_watl2, 1, function(x) {
+    res <- t.test(x[high], x[low], mu = 0)
+    c(pvalue = res$p.value,
+      M = unname(res$estimate[1] - res$estimate[2]))
```

Comment:
I'm just also directly reporting the difference in means so there is no need for an extra loop/calculation. The name `M` is historical: in microarray experiements the difference in means was always called `M` because in log2 scale you subtract (use *M*inus for the calculation) the means to represent a ratio. So, this M is now our estimate for "differential abundance".